components/match2/commons/opponent.lua

```diff
@@ -0,0 +1,331 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:Opponent
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+local Flags = require('Module:Flags')
+local String = require('Module:StringUtils')
+local Table = require('Module:Table')
+local TeamTemplate = require('Module:TeamTemplate')
+
+--[[
+Structural type representation of an opponent.
+
+Examples:
+{type = 'solo', players = {displayName = 'Neeb'}}
+{type = 'team', template = 'alpha x 2020'}
+{type = 'literal', name = 'B2'}
+
+See Opponent.isOpponent for the exact encoding scheme and required fields.
+
+- For opponent display components, see Module:OpponentDisplay.
+- For input from wikicode, use {{1Opponent|...}}, {{TeamOpponent|...}} etc.
+- Used by: PrizePool, GroupTableLeague, match2 Matchlist/Bracket,
+StarcraftMatchSummary
+- Wikis may add additional wiki-specific fields to the opponent representation.
+
+]]
+local Opponent = {}
+
+Opponent.partyTypes = {'solo', 'duo', 'trio', 'quad'}
+Opponent.types = Array.extend(Opponent.partyTypes, {'team', 'literal'})
+
+Opponent.partySizes = {
+	solo = 1,
+	duo = 2,
+	trio = 3,
+	quad = 4,
+}
+
+function Opponent.typeIsParty(type)
```

Comment:
```suggestion
function Opponent.isParty(type)
```

Reply:
I named it `typeIsParty` instead of `isParty` because the input is an opponent type instead of an opponent.

`Opponent.isParty` would be:
```
function Opponent.isParty(type)
    return Opponent.typeIsParty(opponent.type)
end
```